### PR TITLE
Add trait bound `MaybeSerializeDeserialize` for AssetId

### PR DESCRIPTION
**Pull request by en**
so it can work with #[pallet::genesis_config]

### CODE REVIEW

File: frame/support/src/traits/tokens/misc.rs [hunk 1]
```diff
@@ -161,8 +161,8 @@ impl WithdrawReasons {
 }
 
 /// Simple amalgamation trait to collect together properties for an AssetId under one roof.
-pub trait AssetId: FullCodec + Copy  + Eq + PartialEq + Debug {}
-impl<T: FullCodec + Copy + Eq + PartialEq + Debug> AssetId for T {}
+pub trait AssetId: FullCodec + MaybeSerializeDeserialize + Copy  + Eq + PartialEq + Debug {}
```
Comment (xlc) [inline frame/support/src/traits/tokens/misc.rs:164]: Maybe not here? It is a pallet specific thing.

Comment (en) [reply frame/support/src/traits/tokens/misc.rs:164]: Hi Bryan,

With this modification, we can combine `AssetId` (in pallet_assets) with other more abstract types in  `frame_support::traits::tokens`.

We could define a type in our pallet and impl our trait like this:
```
type AssetIdOf<T> =
	<<T as Config>::Assets as fungibles::Inspect<<T as frame_system::Config>::AccountId>>::AssetId;

pub trait Config
{
	type Assets: fungibles::Mutate<<Self as frame_system::Config>::AccountId>;
}

impl pallet_example::Config for Runtime {
	type Assets = Assets;
}
```

Without the trait bound, the following code would not compile:
```
#[pallet::genesis_config]
pub struct GenesisConfig<T: Config> {
	pub asset_id: Vec<(String, AssetIdOf<T>)>,
}
```

Comment (xlc) [reply frame/support/src/traits/tokens/misc.rs:164]: There are ways to workaround this issue.

e.g.

```
type AssetId = u32;

pub trait Config
{
	type Assets: fungibles::Mutate<<Self as frame_system::Config>::AccountId, AssetId>;
}
```

Or if you do want to make AssetId configurable

```
pub trait Config
{
    type AssetId = // anything you want;
    type Assets: fungibles::Mutate<<Self as frame_system::Config>::AccountId, AssetId = Self::AssetId>;
}
```

Comment (en) [reply frame/support/src/traits/tokens/misc.rs:164]: Thanks! That is a better way to do it, I'll close this pr later.

Comment (juanitalaw) [reply frame/support/src/traits/tokens/misc.rs:164]: > Maybe not here? It is a pallet specific thing.

https://github.com/paritytech/substrate/pull/9187#discussion_r657524578

**Pull request closed without merging**